src/develop/develop.c

```diff
@@ -348,6 +349,9 @@ void dt_dev_process_preview_job(dt_develop_t *dev)
 
   dev->preview_status = DT_DEV_PIXELPIPE_VALID;
 
+  if(!dev->history_postpone_invalidate)
+    dt_image_update_final_size(dev->preview_pipe->output_imgid); // move to dt_dev_process_preview_job?
```

Comment:
not sure the comment is still relevant as we *are* `dt_dev_process_peview_job`

Reply:
:-)

I meant of course `dt_dev_process_image_job`; for some reason the existing handling of metadata connected to the `DT_SIGNAL_DEVELOP_PREVIEW_PIPE_FINISHED` rather than the `DT_SIGNAL_DEVELOP_UI_PIPE_FINISHED` signal, so I put the handling where that signal is sent, but it might make more sense to update when the central image is finished rather than the preview image. I'm only now wondering what would happen if the small preview (top left) is hidden; presumably this job doesn't run at all then and metadata doesn't get updated?

A quick check proves that it still works even if navigation pane hidden, so presumably that job always runs, even if the result is never shown. It is possible that the lower resolution preview is finished calculating earlier, so that metadata update could happen a little faster too, although I haven't been able to confirm that visually.